spec.html

```diff
@@ -31055,14 +31055,23 @@ <h1>get RegExp.prototype.dotAll</h1>
         <p>`RegExp.prototype.dotAll` is an accessor property whose set accessor function is *undefined*. Its get accessor function performs the following steps:</p>
         <emu-alg>
           1. Let _R_ be the *this* value.
-          1. If Type(_R_) is not Object, throw a *TypeError* exception.
-          1. If _R_ does not have an [[OriginalFlags]] internal slot, then
-            1. If SameValue(_R_, %RegExp.prototype%) is *true*, return *undefined*.
-            1. Otherwise, throw a *TypeError* exception.
-          1. Let _flags_ be _R_.[[OriginalFlags]].
-          1. If _flags_ contains the code unit 0x0073 (LATIN SMALL LETTER S), return *true*.
-          1. Return *false*.
+          1. Let _cu_ be the code unit 0x0073 (LATIN SMALL LETTER S).
+          1. Return ? RegExpHasFlag(_R_, _cu_).
         </emu-alg>
+
+        <emu-clause id="sec-regexphasflag" aoid="RegExpHasFlag">
```

Comment:
Seems a bit odd to nest this under `dotAll`, but I guess it's fine.

Reply:
> Seems a bit odd to nest this under `dotAll`

Yeah, I agree. I'm not thrilled with its placement, but there doesn't seem to be a good place for it. Personally, I'd prefer that the definition appear after *all* uses (so, nested under `unicode`), but my impression (though I haven't done a thorough check) is that the spec has a convention, in cases like this, for nesting after the *first* use.

Alternatively, we could create a 22.2.8 clause for RegExp AOs: RegExpHasFlag, RegExpExec, RegExpBuiltinExec, AdvanceStringIndex.

Reply:
15 months later, #2843 :-)